File: scripts/screenshot_audit.js
#!/usr/bin/env node

/**
 * screenshot_audit.js — Captura screenshots de cada aba do dashboard e compara
 *
 * Uso:
 *   node scripts/screenshot_audit.js
 *
 * Saída: screenshots em dashboard/audit-screenshots/{timestamp}/
 */

const { chromium } = require('@playwright/test');
const fs = require('fs');
const path = require('path');

const TABS = ['now', 'portfolio', 'performance', 'fire', 'withdraw', 'backtest', 'macro'];
const SCREENSHOT_DIR = path.join(__dirname, '..', 'dashboard', 'audit-screenshots');
const ARCHIVE_DIR = path.join(SCREENSHOT_DIR, new Date().toISOString().split('T')[0]);

// Criar diretório se não existir
if (!fs.existsSync(ARCHIVE_DIR)) {
  fs.mkdirSync(ARCHIVE_DIR, { recursive: true });
}

(async () => {
  let browser;
  try {
    browser = await chromium.launch({
      headless: true,
      args: ['--disable-blink-features=AutomationControlled']
    });

    const context = await browser.newContext({
      viewport: { width: 1440, height: 900 },
      deviceScaleFactor: 1
    });

    const page = await context.newPage();

    // Navegue para o dashboard
    const dashboardUrl = 'file://' + path.join(__dirname, '..', 'index.html');
    console.log(`📸 Carregando dashboard: ${dashboardUrl}`);
    await page.goto(dashboardUrl, { waitUntil: 'networkidle' });

    // Espere o dashboard carregar completamente
    await page.waitForSelector('[data-tab]', { timeout: 10000 }).catch(() => {
      console.warn('⚠️  Tabs não encontrados com data-tab, tentando por ID');
    });

    // Tire screenshots de cada aba
    for (const tab of TABS) {
      try {
        console.log(`📸 Capturando ${tab}...`);

        // Clique na aba (tente múltiplos seletores)
        const selectors = [
          `[data-tab="${tab}"]`,
          `button[id*="${tab}"]`,
          `a[href*="${tab}"]`,
          `.tab-${tab}`,
          `[role="tab"][aria-label*="${tab}"]`
        ];

        let clicked = false;
        for (const selector of selectors) {
          if (await page.locator(selector).first().isVisible().catch(() => false)) {
            await page.click(selector);
            clicked = true;
            break;
          }
        }

        if (!clicked && tab !== 'now') {
          console.warn(`  ⚠️  Aba ${tab} não encontrada, pulando...`);
          continue;
        }

        // Aguarde renderização
        await page.waitForTimeout(1000);

        // Tire screenshot
        const filename = `${tab.padStart(2, '0')}-${tab}-tab.png`;
        const filepath = path.join(ARCHIVE_DIR, filename);
        await page.screenshot({ path: filepath, fullPage: false });

        console.log(`  ✅ ${filename} salvo`);
      } catch (err) {
        console.error(`  ❌ Erro ao capturar ${tab}: ${err.message}`);
      }
    }

    console.log(`\n✅ Screenshots salvos em: ${ARCHIVE_DIR}`);
    console.log(`\n📊 Próximas etapas:`);
    console.log(`  1. Revisar screenshots em ${ARCHIVE_DIR}`);
    console.log(`  2. Comparar com versão anterior (se existir)`);
    console.log(`  3. Validar regressions visuais`);

    await context.close();
  } catch (err) {
    console.error('❌ Erro:', err);
    process.exit(1);
  } finally {
    if (browser) {
      await browser.close();
    }
  }
})();

File: scripts/screenshot_audit.py
#!/usr/bin/env python3
"""
screenshot_audit.py — Captura e compara screenshots do dashboard usando html2canvas

Uso:
    python3 scripts/screenshot_audit.py
"""

import json
import os
import subprocess
import sys
from pathlib import Path
from datetime import datetime

ROOT = Path(__file__).parent.parent
DASHBOARD_HTML = ROOT / "index.html"
SCREENSHOT_DIR = ROOT / "dashboard" / "audit-screenshots"
ARCHIVE_DIR = SCREENSHOT_DIR / datetime.now().strftime("%Y-%m-%d_%H-%M-%S")

# Criar diretório
ARCHIVE_DIR.mkdir(parents=True, exist_ok=True)

# Criar HTML auxiliar que captura screenshots via html2canvas
CAPTURE_HTML = f"""
<!DOCTYPE html>
<html>
<head>
    <meta charset="UTF-8">
    <script src="https://cdnjs.cloudflare.com/ajax/libs/html2canvas/1.4.1/html2canvas.min.js"></script>
    <style>
        body {{ margin: 0; padding: 10px; background: #f5f5f5; }}
        iframe {{ width: 100%; height: 900px; border: 1px solid #ccc; margin: 20px 0; }}
        button {{ padding: 10px 20px; margin: 5px; cursor: pointer; }}
    </style>
</head>
<body>
    <h1>📸 Dashboard Screenshot Audit</h1>
    <p>Capturando screenshots de cada aba...</p>
    <div id="status"></div>

    <iframe id="dashboard" src="file://{str(DASHBOARD_HTML)}"></iframe>

    <script>
        const tabs = ['now', 'portfolio', 'performance', 'fire', 'withdraw', 'backtest', 'macro'];
        const statusDiv = document.getElementById('status');
        const iframe = document.getElementById('dashboard');

        async function captureTab(tabName) {{
            try {{
                const iframeContent = iframe.contentDocument || iframe.contentWindow.document;

                // Clique na aba
                const tabBtn = iframeContent.querySelector(`[data-tab="{tabName}"]`) ||
                              iframeContent.querySelector(`button:contains("{tabName}")`);
                if (tabBtn) tabBtn.click();

                // Aguarde renderização
                await new Promise(r => setTimeout(r, 1500));

                // Capture a aba
                const canvas = await html2canvas(iframeContent.body, {{
                    backgroundColor: '#ffffff',
                    scale: 2,
                    logging: false,
                    useCORS: true,
                    allowTaint: true
                }});

                const link = document.createElement('a');
                link.href = canvas.toDataURL('image/png');
                link.download = `${{tabName}}.png`;
                link.click();

                statusDiv.innerHTML += `<p>✅ Capturado: ${{tabName}}</p>`;
            }} catch (err) {{
                statusDiv.innerHTML += `<p>❌ Erro em ${{tabName}}: ${{err.message}}</p>`;
            }}
        }}

        (async () => {{
            for (const tab of tabs) {{
                await captureTab(tab);
            }}
            statusDiv.innerHTML += '<p>✅ Concluído!</p>';
        }})();
    </script>
</body>
</html>
"""

print("📸 Screenshot Audit — Dashboard")
print(f"Diretório: {ARCHIVE_DIR}")
print("\n⚠️  Abordagem: html2canvas (renderização no navegador)")
print("Para screenshots completos, use:")
print("  - wkhtmltopdf (instale com: apt-get install wkhtmltopdf)")
print("  - Playwright (instale com: pip install playwright && playwright install)")
print("\nAlternativa rápida: use /dashboard-pdf skill para gerar PDF com screenshots")
